src/ds/d500/d500-auto-calibration-handler.h

```diff
@@ -0,0 +1,66 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include "d500-private.h"
+#include <src/hw-monitor.h>
+
+
+namespace librealsense
+{
+    enum class d500_calibration_state
+    {
+        RS2_D500_CALIBRATION_STATE_IDLE = 0,
+        RS2_D500_CALIBRATION_STATE_PROCESS,
+        RS2_D500_CALIBRATION_STATE_DONE_SUCCESS,
+        RS2_D500_CALIBRATION_STATE_DONE_FAILURE,
+        RS2_D500_CALIBRATION_STATE_FLASH_UPDATE,
+        RS2_D500_CALIBRATION_STATE_COMPLETE
+    };
+
+    enum class d500_calibration_result
+    {
+        RS2_D500_CALIBRATION_RESULT_UNKNOWN = 0,
+        RS2_D500_CALIBRATION_RESULT_SUCCESS,
+        RS2_D500_CALIBRATION_RESULT_FAILED_TO_CONVERGE,
+        RS2_D500_CALIBRATION_RESULT_FAILED_TO_RUN
+    };
+
+    enum class d500_calibration_mode
+    {
+        RS2_D500_CALIBRATION_MODE_RESERVED = 0,
+        RS2_D500_CALIBRATION_MODE_RUN,
+        RS2_D500_CALIBRATION_MODE_ABORT,
+        RS2_D500_CALIBRATION_MODE_DRY_RUN
+    };
+
+#pragma pack(push, 1)
+    struct d500_calibration_answer
+    {
+        uint8_t calibration_state;
+        int8_t calibration_progress;
+        uint8_t calibration_result;
+        ds::d500_coefficients_table depth_calibration;
+    };
+#pragma pack(pop)
+
+    class d500_device;
+    class debug_interface;
+
+    class d500_auto_calibrated_handler
```

Comment:
Not really a handler, is it? It's an implementation of the different generic TC APIs (run/get-status/get-config/set-config).
If we wanted to replace this with a DDS implementation, how would we do it?
Wouldn't we want this to be an implementation of a generic interface that we can derive from to implement over DDS?

Reply:
I would add a generic `auto_calibrated_camera_interface`.
With the D500-specific `d500_auto_calibrated_camera`?

Reply:
done